File: sort/bubblesort/bubblesort.py

'''
		Time complexity  : O(n^2)
		Space complexity : O(1)
'''

def bubblesort(array) :

	for i in range(len(array) + 1) :
		for j in range(0 , len(array) - i - 1) :
			if array[j] > array[j + 1] :
				array[j], array[j+1] = array[j+1], array[j]
	return array

a = [8 , 2 , 5 , 7 , 0 , 9 , 4 , 6 , 1 , 3]
print(bubblesort(a))

File: sort/bubblesort/bubblesort.cpp

#include <bits/stdc++.h>

using namespace std ;

void bubblesort(int arr[] , int n)
{
	/*
		Time complexity  : O(n^2)
		Space complexity : O(1)
	*/

	for(int i = 0 ; i < n ; i++)
	{
		for(int j = 0 ; j < n - 1 ; j++)
		{
			if(arr[j] > arr[j + 1])
			{
				swap(arr[j] , arr[j +1]) ;
			}
		}
	}

	cout << "[ " ;
	for(int i = 0 ; i < n ; i++)
	{
		cout << arr[i] << " " ;
	}
	cout << "]" << endl ;
}

int main(void)
{
	int arr[] = {4 , 8 , 6 , 1 , 0 , 5 , 9 , 7 , 3 , 2} ;
	bubblesort(arr , sizeof(arr) / sizeof(arr[0])) ;
}